### DISCUSSION TITLE
Do you guys offer an API to access to be able to change values outside of the website?

### DISCUSSION THREAD

**Discussion by PaulHaze** [evilmartians/oklch-picker#146] [General]
So I've recently discovered the OKLCH color space as a result of working with the DaisyUI framework and the benefits of it are clear. 

I've noticed there are no extensions within VS Code that enable conversions of colors into the OKLCH space, and I thought as a nice challenge I would explore what it would take to to build one. As such I was wondering if there was an API that your site offers that would enable me to say, end over a HEX or a RGB value and get back the OKLCH value? I would love to be able to right click a color within VS Code, choose 'Convert color to OKLCH' from the context menu and have it automatically change the colour....

EDIT: Straight after posting this I have come across [THIS](https://gist.github.com/dkaraush/65d19d61396f5f3cd8ba7d1b4b3c9432) gist which seems to do what I am after so I will play around with this first....

**Comment by ai** [comment]
Using external API for hex→oklch convert is not the best way to design VS Code plugin.

Just convert it locally on the client’s machine.

I recommend [`culori`](https://culorijs.org) (which I use) or [`Color.js`](https://colorjs.io).

```js
import { oklch, parse, format } from 'culori'

formatCss(oklch(parse(hexColor)))
```

That gist will work too, it looks exactly what `culori` does inside.
